common/persistence/wrappers/metered/base.go

```diff
@@ -143,6 +143,23 @@ func (p *base) call(scope int, op func() error, tags ...metrics.Tag) error {
 	return err
 }
 
+func (p *base) callWithoutDomainTag(scope int, op func() error, tags ...metrics.Tag) error {
```

Comment:
what about the functions that uses `callWithDomainAndShardScope`? category will be missing for those

Reply:
These history tasks methods don't fall into that branch, because they are not [domainTaggedRequest](https://github.com/cadence-workflow/cadence/blob/master/common/persistence/wrappers/metered/base.go#L309).

Reply:
Got it. So some of the generated method never call that branch which uses `callWithDomainAndShardScope` and all history task related methods are in that category. Would be nice to change code gen template to avoid spitting unused code into each function.